gstreamer-sdp/src/s_d_p_time.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 2018 Sebastian Dröge <sebastian@centricular.com>
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::mem;
+use std::ffi:CStr;
+
+use ffi;
+
+use auto::SDPResult;
+
+pub struct SDPTime(ffi::GstSDPTime);
+
+impl SDPTime {
+    pub fn new(start: &str, stop: &str, repeat: &[&str]) -> Result<Self, SDPResult> {
+        assert_initialized_main_thread!();
+        unsafe {
+            let mut time = mem::uninitialized();
+            let result =from_glib(ffi::gst_sdp_time_set(
+                &mut time,
+                start.to_glib_none().0,
+                stop.to_glib_none().0,
+                repeat.to_glib_none().0,
+            ));
+			match result {
+				SDPResult::Ok => Ok(time),
+				_ => Err(result),
+			}
+        }
+    }
+
+    pub fn start(&self) -> &str {
+        CStr::from_ptr(self.0.start).to_str().unwrap()
+    }
+
+    pub fn stop(&self) -> &str {
+        CStr::from_ptr(self.0.stop).to_str().unwrap()
+    }
+
+    // TODO: How to convert glib::GArray into a string slice?
```

Comment:
I wasn't sure how to convert `glib::GArray` into a string slice. It seems like it _might_ be possible to use `std::slice::from_raw_parts`?

Reply:
Is it an actual string or what is it?
If it is a string, you can use CStr on the data pointer of the GArray.

Reply:
It seems to be an array of `GChars`

Reply:
It's an array of arbitrary data, the question is what GstSDPMessage is exactly putting in there. Which seem to be an array of strings, not a single string. So you could return a `Vec<&str>` here

Reply:
How would I convert this https://docs.rs/glib-sys/0.5.0/glib_sys/struct.GArray.html into a `Vec<&str>`? It have the length there but I don't think I can do something like pointer arithmetic to iterate over all the data elements.

Reply:
```rust
let arr = (*vec).data as *const *const c_char;
let vec = Vec::with_capacity((*arr).len as usize);
for i in 0..len {
    vec.push(CStr::new(arr.offset(i)));
}
```

or something like that

Reply:
Ahhh didn't know about `offset`, thanks @sdroege !